scheduler.py

```diff
@@ -30,23 +34,12 @@ def assign_task(self):
     def get_task_completion(self, task):
         pass
 
-    def start_server(self):
-        server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        server_socket.bind((self.host, self.port))
-        server_socket.listen(5)
-        print("Scheduler listening on {}:{}".format(self.host, self.port))
-        while True:
-            client_socket, addr = server_socket.accept()
```

Comment:
Current implementation is a pull based approach where worker listens for tasks. Need clarity on push/pull method where push only needs 1 api call to submit task to worker and get a future response vs 2 calls for pull based flow.

Reply:
if we can do a push method with grpc, I think that would be preferred.
I did not see a good way of implementing push method using socket programming hence it is programmed this way.